### PR TITLE
[kube-prometheus-stack] dynamic endpoint configuration according to k8s version

### LINKED ISSUE #2079: [kube-prometheus-stack] kube-controller-manager and kube-scheduler endpoints are not configured automatically according to targeted k8s version
### Describe the bug a clear and concise description of what the bug is.

@invidian did a great job in https://github.com/prometheus-community/helm-charts/pull/1393 but today I discovered that when you specify endpoints for kubeControllerManager and kubeScheduler, for a k8s cluster provisioned with eg Rancher, where these services doesn't run as a pod but as a docker service on the host, you get an error that the required port is not defined.

A workaround is to specify the port manually in the values file, eg

```
kubeControllerManager:
  endpoints:
    - 10.10.10.10
    - 10.10.10.11
    - 10.10.10.12
  service:
    port: 10257
```

### What's your helm version?

version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.16.9"}

### What's your kubectl version?

Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.9", GitCommit:"6df4433e288edc9c40c2e344eb336f63fad45cd2", GitTreeState:"clean", BuildDate:"2022-04-13T19:52:02Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

Kube-prometheus-stack

### What's the chart version?

35.2.0

### What happened?

Errors out with a dry-run error that a required endpoint port is not defined.

### What you expected to happen?

Not having to specify the port and make it dynamic based on detected k8s version.

### How to reproduce it?

Deploy the chart specifying endpoints but not the service port, eg

```
kubeControllerManager:
  endpoints:
    - 10.10.10.10
    - 10.10.10.11
    - 10.10.10.12
```

### Enter the changed values of values.yaml?

_No response_

### Enter the command that you execute and failing/misfunctioning.

ArgoCD takes care of deploying so don't know the exact command it uses.

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by jabbors**
#### What this PR does / why we need it:

Dynamically configures the port for kube-controller-manager and
kube-scheduler endpoints according to targeted k8s version.

#### Which issue this PR fixes
- fixes #2079

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 35.5.1
+version: 35.5.2
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]:
```suggestion
version: 35.5.3
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by jabbors** [comment]
Thanks for keeping the PR up-to-date. Is there anything I can do to speed up an approval and merge?

**Review by monotek — CHANGES_REQUESTED** [review]
seems to be oboslte as https://github.com/prometheus-community/helm-charts/pull/2162 was merged

**Comment by jabbors** [comment]
yeah, then we can close this PR

**Pull request closed without merging**